integration/experiment/power_sweep/gen_policy_recommendation.py

```diff
@@ -0,0 +1,287 @@
+        x1 = np.array(PL_list).reshape((-1, 1))
+        yy = np.array(value_list)
+        xn = self._transform.fit_transform(x1)
+        self._model = LinearRegression().fit(xn, yy)
+
+    def test(self, PL):
+        # TODO shouldn't have to look inside the coeffs and 
+        # intercept to do this
+        return self._model.intercept_ + \
+               sum([
+                   c * PL**(i+1) for i, c in enumerate(self._model.coef_)
+                   ])
+
+    def __str__(self):
+        if self._model:
+            terms = ["", " PL"] + \
+                    [" PL ** %d" % (i+1)
+                            for i in range(1, len(self._model.coef_))]
+            coeffs = [self._model.intercept] + self._model.coef_
+            expression = ""
+            terms.reverse() ; coeffs.reverse() ; firstTerm = True
+            for tt, cc in zip(terms, coeffs):
+                if not firstTerm:
+                    if cc < 0:
+                        expression += " - "
+                        cc *= -1
+                    else:
+                        expression += " + "
+                expression += str(cc) + tt
+            # TODO classname
+            return "<%s %s>" % ("Polyfitmodel", expression)
+        else:
+            return "<%s (untrained)>" % "Polyfitmodel"  #TODO classname
+
+
+class CubicPowerModel(PolynomialFitPowerModel):
+    def __init__(self):
+        super().__init__(degree=3)
+
+
+def extract_from_reports(path, region_filter):
+    simplified_data = []
+    
+    try:
+        df = geoio.RawReportCollection("*report", dir_name=path).get_df()
+    except RuntimeError:
+        sys.stderr.write("<geopm> Error: No report data found in " + path + \
+                "; run a power sweep before using this analysis.\n")
+        sys.exit(1)
+    for profile in df['Profile'].unique():
+        # profile will be something like "unifiedmodel_power_governor_250_1"
+        power_limit = int(profile.split('_')[-2])  # TODO - access the policy
+        df_profile = df[df['Profile'] == profile]
+        for host in df_profile['host'].unique():
+            df_profile_host = df_profile[df_profile['host'] == host]
+            total_rt = 0
+            total_en = 0
+            for region_name in df_profile_host['region'].unique():
+                if region_filter and region_name not in region_filter:
+                    continue
+                region = df_profile_host[df_profile_host['region'] == region_name]
+                runtime = sum(region['runtime (sec)'])
+                energy = sum(region['package-energy (joules)'])
+    
+                total_rt += runtime
+                total_en += energy
+    
+            simplified_data.append((power_limit, total_rt, total_en))
+    return simplified_data
+
+
+def dump_simplified_data(simplified_data, fname):
+    raw_out = open(fname, "w")
+    for pl, rt, en in sorted(simplified_data):
+        raw_out.write("%d %f %f\n" % (pl, rt, en))
+    raw_out.close()
```

Comment:
This should be refactored to use a ```with``` statement.  See: https://github.com/geopm/geopm/blob/dev/integration/experiment/power_sweep/gen_plot_balancer_comparison.py#L259-L260

Reply:
Done.